compiler/reports.nim

```diff
@@ -115,9 +287,7 @@ func severity*(rep: LexerReport): ReportSeverity =
     else: rsevTrace
 
 type
-  ParserReportKind* = enum
-    rparName = "Name" ## Linter report about used identifier
-
+  ParserReportKind* = range[rparInvalidIndentation .. rparName]
   ParserReport* = object of ReportBase
     kind*: ParserReportKind
```

Comment:
This requires disabling check for the nonzero discriminant value. In short, I want to try and use this pattern for reports. 

```nim
type
  E = enum a, b, c, d
  E2 = range[c .. d]
  Ob = object
    case k*: E2
      of c: discard
      of d: discard
```

Having multiple enums for different kinds of reports is not sustainable, too much friction. Having a single enum that is split into ranges makes things a lot easier. 

An alternative would be to merge all report kinds into a single variant object, but then I'm afraid we will simply run out of meaningful field names, or will end up with another `TSym`-like object that contains absolutely everything mixed into one pot, with no separation whatsoever.

Reply:
https://github.com/nim-works/nimskull/pull/99